src/sys/signal.rs

```diff
@@ -722,6 +735,25 @@ mod tests {
         assert_eq!(action_ign.handler(), SigHandler::SigIgn);
     }
 
+    #[test]
+    fn test_procalive() {
+        use sys::wait::waitpid;
+        use unistd::fork;
+        use unistd::ForkResult::*;
+
+        match fork() {
+            Ok(Child) => loop {},
+            Ok(Parent { child }) => {
+                assert_eq!(proc_alive(child).unwrap(), true);
+                kill(child, SIGTERM).unwrap();
```

Comment:
Signal handlers are inherited across a fork, so any test that installs a `SIGTERM` handler will screw up this test.  You should therefore grab the `SIGNAL_MTX` at the top of this test.  If you grab multiple mutexes, make sure to do it in alphabetical order to prevent deadlocks.